pkg/api/v1/conversion_test.go

```diff
@@ -101,3 +102,54 @@ func TestListOptionsConversion(t *testing.T) {
 		}
 	}
 }
+
+func TestResourceListConversion(t *testing.T) {
+	bigMilliQuantity := resource.NewQuantity(resource.MaxMilliValue, resource.DecimalSI)
+	bigMilliQuantity.Add(resource.MustParse("1m"))
+	bigRoundedQuantity := resource.NewQuantity(resource.MaxMilliValue+1, resource.DecimalSI)
+
+	tests := []struct {
+		input    versioned.ResourceList
+		expected api.ResourceList
+	}{
+		{ // No changes necessary.
+			input: versioned.ResourceList{
+				versioned.ResourceMemory:  resource.MustParse("30M"),
+				versioned.ResourceCPU:     resource.MustParse("100m"),
+				versioned.ResourceStorage: resource.MustParse("1G"),
+			},
+			expected: api.ResourceList{
+				api.ResourceMemory:  resource.MustParse("30M"),
+				api.ResourceCPU:     resource.MustParse("100m"),
+				api.ResourceStorage: resource.MustParse("1G"),
+			},
+		},
+		{ // Nano-core CPU value should be rounded up to millicores.
+			input: versioned.ResourceList{
+				versioned.ResourceCPU: resource.MustParse("3000023n"),
+			},
+			expected: api.ResourceList{
+				api.ResourceCPU: resource.MustParse("4m"),
+			},
+		},
+		{ // Large CPU value should be rounded up to cores.
```

Comment:
I think that CPU should always have millicore precision, no more and no less.  If you wan to make a new api.ResourceFoo and have different precision, then go ahead and do that.